yt_dlp/extractor/zdf.py

```diff
@@ -349,108 +425,194 @@ class ZDFIE(ZDFBaseIE):
+            })
+
+        return {
+            'thumbnails': thumbnails,
+            **traverse_obj(video, {
+                'title': ('document', 'titel', {str}),
+                'description': ('document', 'beschreibung', {str}),
+                'timestamp': (
+                    (('document', 'date'), ('meta', 'editorialDate')),
+                    {unified_timestamp}, any),
+                'subtitles': ('document', 'captions', {self._extract_subtitles}),
+            }),
+            **self._extract_ptmd(ptmd_url, document_id, self._get_api_token()),
+            'id': document_id,
+        }
+
     def _real_extract(self, url):
         video_id = self._match_id(url)
+        video_data = self._download_graphql(video_id, 'video metadata', body={
+            'operationName': 'VideoByCanonical',
+            'query': self._GRAPHQL_QUERY,
+            'variables': {'canonical': video_id},
+        })['data']['videoByCanonical']
+
+        if not video_data:
+            return self._extract_fallback(video_id)
+
+        aspect_ratio = None
+        ptmd_urls = []
+        for node in traverse_obj(video_data, ('currentMedia', 'nodes', lambda _, v: v['ptmdTemplate'])):
+            ptmd_url = self._expand_ptmd_template('https://api.zdf.de', node['ptmdTemplate'])
+            # Smuggle vod_media_type so that _extract_ptmd is aware of 'DGS' variants
+            if vod_media_type := node.get('vodMediaType'):
+                ptmd_url = smuggle_url(ptmd_url, {'vod_media_type': vod_media_type})
+            ptmd_urls.append(ptmd_url)
+            if not aspect_ratio:
+                aspect_ratio = self._parse_aspect_ratio(node.get('aspectRatio'))
 
-        webpage = self._download_webpage(url, video_id, fatal=False)
-        if webpage:
-            player = self._extract_player(webpage, url, fatal=False)
-            if player:
-                return self._extract_regular(url, player, video_id, query={'profile': 'player-3'})
-
-        return self._extract_mobile(video_id)
+        return {
+            **traverse_obj(video_data, {
+                'title': ('title', {str}),
+                'description': (('leadParagraph', ('teaser', 'description')), any, {str}),
+                'timestamp': ('editorialDate', {parse_iso8601}),
+                'thumbnails': ('teaser', 'image', 'list', {self._extract_thumbnails}),
+                'episode_number': ('episodeInfo', 'episodeNumber', {int_or_none}),
+                'season_number': ('episodeInfo', 'seasonNumber', {int_or_none}),
+                'series': ('smartCollection', 'title', {str}),
+                'series_id': ('smartCollection', 'canonical', {str}),
+                'chapters': ('currentMedia', 'nodes', 0, 'streamAnchorTags', 'nodes', {self._extract_chapters}),
+            }),
+            **self._extract_ptmd(ptmd_urls, video_id, self._get_api_token(), aspect_ratio),
+            'id': video_id,
+        }
 
 
 class ZDFChannelIE(ZDFBaseIE):
```

Comment:
Let's do this for the time being; we don't have pagination, and the site doesn't use the `GetSmartCollectionByCanonical` operation for all types of playlists, e.g. `seasonByCanonical` is used for `serien` URLs. This can wait for another PR, I'd like to merge what we've got
```suggestion
class ZDFChannelIE(ZDFBaseIE):
    _WORKING = False
```

Reply:
The endpoint is used by the mobile app and then additional entries are paginated in via a different endpoint if the list is long enough or if you switch seasons. I suspect the webpage may do the same to get the information that is hidden in the hydration data. But if so, it obviously does it server-side where we can't see it.

So I assumed it is fine to use the `GetSmartCollectionByCanonical` and I just cranked up the limit from 24 (used in the app) to 100 (server-enforced limit). This already gets you everything there is for 95%+ of cases I would guess.

So I though it would be enough to have a warning for the edge case for the time being and the extractor would not need to be marked as broken. Is there any real user-facing consequence of marking it as broken? I know tests don't run and it is marked as broken in the list of supported sites. But those are relatively low impact.

Reply:
Nevermind, I fixed it with pagination and all. We're now using a combination of the mobile app endpoint and the website endpoint